### PR TITLE
[Dependencies] Remove `kubernetes` requirement

### LINKED ISSUE #36331: [Core] Remove `kubernetes` from dependencies
### Description

Currently `ray[all]` installs `ray[k8s]` which in particular installs `kubernetes`: 

https://github.com/ray-project/ray/blob/master/python/setup.py#L267

This package is too big and should be removed from the dependencies.

### Use case

_No response_

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
The `kubernetes` package is installed with `ray["all"]` but it's currently only used in two places:

1. A helper function for converting units
2. The GCS FT on K8s release test: https://github.com/ray-project/ray/blob/407062499038344b0f3edb54b2c7985c3320d1ec/release/k8s_tests/run_gcs_ft_on_k8s.py#L4

This PR removes `kubernetes` from Ray's dependencies and removes it from the Ray docker image. This change is needed to reduce the overall package size of `ray["all"]`.  

For the helper function, we vendor it from the upstream.

For the release test, `kubernetes` is installed separately: https://github.com/ray-project/ray/blob/407062499038344b0f3edb54b2c7985c3320d1ec/release/k8s_tests/app_config.yaml#L13.  I've also added it to `release/requirements.txt` to be safe.

## Related issue number

Closes #36331 
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by architkulkarni** [comment]
Assigning @iycheng as a codeowner from Ray Core

File: python/ray/autoscaler/_private/kuberay/utils.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Source: 
+# https://github.com/kubernetes-client/python/blob/master/kubernetes/utils/quantity.py
+from decimal import Decimal, InvalidOperation
+
+
```
Comment (ijrsvt) [inline python/ray/autoscaler/_private/kuberay/utils.py:5]: This LGTM as this code has remained pretty much static in the upstream.

**Review by ijrsvt — APPROVED** [review]
LGTM!

**Review by fishbone — APPROVED** [review]

**Review by edoakes — APPROVED** [review]
stamping dependency change

**Comment by architkulkarni** [comment]
huggingface_text_classification broken on master, unrelated
`rllib:learning_tests_two_step_game_qmix` flaky on master, unrelated
RLlib memory leak failure Error in download: java.io.IOException: Error downloading [https://dl.google.com/go/go1.18.linux-amd64.tar.gz]  unrelated
RLlib failure nested_action_spaces_ppo_torch unrelated
Build images failure flaky on master

**Pull request merged**